src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2837,6 +2847,27 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         e.DragUIOverride().IsGlyphVisible(false);
     }
 
+    // Method description:
+    // - Checks if the uri is valid and sends an event if so
+    // Arguments:
+    // - The uri
+    void TermControl::_HyperlinkHandler(std::wstring uri)
+    {
+        try
+        {
+            auto parsed = winrt::Windows::Foundation::Uri(uri);
+            if (parsed.SchemeName() == L"http" || parsed.SchemeName() == L"https")

```

Comment:
I might suggest moving the validation up another layer -- just call `_openHyperlinkHandlers` and let TerminalApp make the final decision on whether http/https/file is okay